src/vcpkg/commands.test-features.cpp

```diff
@@ -50,6 +53,66 @@ namespace
+
+    std::vector<std::string> get_for_merge_with_test_port_names(const VcpkgPaths& paths, StringView for_merge_with)
+    {
+        auto& fs = paths.get_filesystem();
+        auto& builtin_ports = paths.builtin_ports_directory();
+        auto git_exe = paths.get_tool_exe(Tools::GIT, out_sink);
+        auto ports_dir_prefix =
+            git_prefix(console_diagnostic_context, git_exe, builtin_ports).value_or_quiet_exit(VCPKG_LINE_INFO);
+        const auto locator = GitRepoLocator{GitRepoLocatorKind::CurrentDirectory, builtin_ports};
+        auto index_file =
+            git_index_file(console_diagnostic_context, git_exe, locator).value_or_quiet_exit(VCPKG_LINE_INFO);
+        TempFileDeleter temp_index_file{fs, fmt::format("{}_vcpkg_{}.tmp", index_file, get_process_id())};
+        if (!fs.copy_file(
+                console_diagnostic_context, index_file, temp_index_file.path, CopyOptions::overwrite_existing) ||
+            !git_add_with_index(console_diagnostic_context, git_exe, builtin_ports, temp_index_file.path))
+        {
+            Checks::exit_fail(VCPKG_LINE_INFO);
+        }
+        auto head_tree = git_write_index_tree(console_diagnostic_context, git_exe, locator, temp_index_file.path)
+                             .value_or_quiet_exit(VCPKG_LINE_INFO);
+        auto merge_base = git_merge_base(console_diagnostic_context, git_exe, locator, for_merge_with, "HEAD")
+                              .value_or_quiet_exit(VCPKG_LINE_INFO);
+        auto diffs = git_diff_tree(console_diagnostic_context,
+                                   git_exe,
+                                   locator,
+                                   fmt::format("{}:{}", merge_base, ports_dir_prefix),
+                                   fmt::format("{}:{}", head_tree, ports_dir_prefix))
+                         .value_or_quiet_exit(VCPKG_LINE_INFO);
+        std::vector<std::string> test_port_names;
+        for (auto&& diff : diffs)
+        {
+            switch (diff.kind)
+            {
+                case GitDiffTreeLineKind::Added:
+                case GitDiffTreeLineKind::Copied:
+                case GitDiffTreeLineKind::Modified:
+                case GitDiffTreeLineKind::Renamed:
+                case GitDiffTreeLineKind::TypeChange: test_port_names.push_back(std::move(diff.file_name)); break;
+                case GitDiffTreeLineKind::Deleted:
+                case GitDiffTreeLineKind::Unmerged:
+                case GitDiffTreeLineKind::Unknown: break;
+                default: Checks::unreachable(VCPKG_LINE_INFO);
+            }
+        }
+
+        msg::print(msg::format(msgForMergeWithTestingTheFollowing, msg::value = for_merge_with)
+                       .append_raw(' ')
+                       .append_raw(Strings::join(" ", test_port_names))
+                       .append_raw('\n'));
+
+        return test_port_names;
+    }
+
+    std::vector<SourceControlFile*> load_all_scf_by_name(std::vector<std::string>&& test_port_names,
```

Comment:
The && is a bit weird since the impl doesn't consume it. Should this be a `View<std::string>`?